### PR TITLE
update the busybox test image to 1.36.1

### PR DESCRIPTION
**Pull request by tzneal**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

There's an odd crash on arm64 in busybox /bin/top using the 1.29-4, 1.32 and 1.33 versions of busybox.  Seems to be resolved in 1.34+.

https://github.com/kubernetes/kubernetes/issues/119584#issuecomment-1653737945

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/images/busybox/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1.29-4
+1.36.1-1
```
[the remark this answers is not in the document]
Comment (tzneal) [reply test/images/busybox/VERSION:1]: I could go either way.  1.36.1-1 is more accurate, but if it causes issues I'm good with 1.36-1 as well.

**Comment by pacoxu** [comment]
Is this target v1.28 or v1.29?

**Comment by dims** [comment]
@pacoxu we are way too late for 1.28, so for 1.29

**Comment by tzneal** [comment]
Yup, agreed. Not sure if I have permissions, but:

/milestone 1.29

**Comment by SergeyKanzhelev** [comment]
quick search revealed another busybox mention here: https://github.com/kubernetes/kubernetes/blob/dfe6685f0f98854e7ab8c78523ca2036e1dbc357/test/images/httpd/BASEIMAGE#L6 Maybe more. Should we add them to dependencies under hack to keep consistent?

**Comment by jaypoulz** [comment]
Would it be possible to backport this into v1.28 and v1.27. We've discovered an issue with the busybox-1.29 image causing problems on linux/ppc64le when running scheduled in privileged containers on a node with an selinux policy that sets selinuxuser_execmod to off.

**Comment by tzneal** [comment]
> Maybe more. Should we add them to dependencies under hack to keep consistent?

I'm not 100% convinced. That particular test image has a mixture of sources:

```
$ cat test/images/httpd/BASEIMAGE
linux/amd64=httpd:2.4.38-alpine
linux/arm=arm32v6/httpd:2.4.38-alpine
linux/arm64=arm64v8/httpd:2.4.38-alpine
linux/ppc64le=ppc64le/httpd:2.4.38-alpine
linux/s390x=s390x/httpd:2.4.38-alpine
windows/amd64/1809=REGISTRY/busybox:1.29-2-windows-amd64-1809
windows/amd64/ltsc2022=REGISTRY/busybox:1.29-2-windows-amd64-ltsc2022
```

the busybox image itself is also a mixture:

```
$ cat test/images/busybox/BASEIMAGE
linux/amd64=busybox:1.29
linux/arm=arm32v6/busybox:1.29
linux/arm64=arm64v8/busybox:1.29
linux/ppc64le=ppc64le/busybox:1.29
linux/s390x=s390x/busybox:1.29
windows/amd64/1809=mcr.microsoft.com/windows/nanoserver:1809
windows/amd64/ltsc2022=mcr.microsoft.com/windows/nanoserver:ltsc2022
```

**Comment by mkumatag** [comment]
> > Would it be possible to backport this into v1.28 and v1.27
> 
> I'm not entirely sure of the policy, but I'll create cherry-pick CRs once this is in and see.

Cherrypick is not required for this code because this is just for building image, we need to update the manifest [here](https://github.com/kubernetes/kubernetes/blob/ad150771936bdd2594b545c6932ae46d8fec4e62/test/utils/image/manifest.go#L241) and cherrypick that one.

**Comment by tzneal** [comment]
I read the criteria at https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md#what-kind-of-prs-are-good-for-cherry-picks and I don't think this would meet the bar for a cherry-pick.

**Comment by jaypoulz** [comment]
@tzneal thanks for checking this! very much appreciated!

**Pull request merged**